File: StorageApi/src/main/kotlin/org/vpp/storageApi/batterySytem/BatterySystemDto.kt
package org.vpp.storageApi.batterySytem

import kotlinx.serialization.Serializable
import org.vpp.storage.batterySystem.SystemStatus
import org.vpp.storageApi.consumerGroup.ConsumerGroupDto
import org.vpp.storageApi.dto.Dto

@Serializable
class BatterySystemDto : Dto() {
    lateinit var status: SystemStatus
    lateinit var manufacturer: String
    lateinit var serialNumber: String
    lateinit var group: ConsumerGroupDto
}

File: internal-frontend/src/app/dto/BatterySystemDto.ts
import {SystemStatus} from "./SystemStatus";
import {ConsumerGroupDto} from "./ConsumerGroupDto";
import {Dto} from "./Dto";

export class BatterySystemDto extends Dto {
  value: number = 0;
  serialNumber: string = '';
  manufacturer: string = '';
  group: ConsumerGroupDto | null = null;
  status: SystemStatus = SystemStatus.ONLINE;
}
